### PR TITLE
fix #3756 prevent modifications by standard operations to user objects

### LINKED ISSUE #3756: Prevent modifications to passed in objects
### Is your task related to a problem? Please describe

Related to #3078 and possibly to https://github.com/java-operator-sdk/java-operator-sdk/issues/830 modifications should not be directly made to passed in objects.

### Describe the solution you'd like

Simply reverting the change is not quite enough.  As the object could have been obtained from a cache that simply creates a narrow race condition.  More than likely the only solution is to clone prior to making the modification.  The impetus for that cloning is mitigated if cloning were enforced for any resource obtained by the informer cache.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by shawkins**
## Description
Guards against the basic operations from modifying passed in resources.

I couldn't see any other places besides the JobOperationsImpl where we making similar modifications.

## Type of change

 - [x] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [x] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [x] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Review by metacosm — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,7 @@
 * Fix #3679: output additionalProperties field with correct value type for map-like fields (CRD Generator)
 * Fix #3648: `Serialization.unmarshal` fails to deserialize YAML with single document in presence of document delimiter(`---`)
 * Fix #3568: Pod file upload fails if the path is `/`
+* Fix #3745: updating require logic to account for item namespace
```
Comment (manusa) [inline CHANGELOG.md:14]: I think this got mixed up with another PR

Comment (shawkins) [reply CHANGELOG.md]: sorry, yes that was from the change specifically for require.  updated now.

**Review by manusa — APPROVED** [review]

**Review by rohanKanojia — APPROVED** [review]

**Comment by manusa** [comment]
Apparently the clone method is not compatible with the usage we are doing for the `additionalProperties` field when target field types don't match the deserialized types.

Checking this now.

**Pull request merged**